bin/serve_update_description_gen/main.go

```diff
@@ -0,0 +1,173 @@
+		Usage: "generate json description file for serve update source of updaterini library",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:     "url",
+				Aliases:  []string{"u"},
+				Required: true,
+				Usage:    "base url for all versions, (base url + version folder + asset file name) is used by library to load asset from server",
+			},
+			&cli.StringFlag{
+				Name:    "descFilename",
+				Aliases: []string{"d"},
+				Value:   descriptionFilename,
+				Usage: "filename in version folder, that contains version description, " +
+					"will be parsed and skipped in assets. Parsed description will be used in result json file",
+			},
+			&cli.StringFlag{
+				Name:    "descNameSeparator",
+				Aliases: []string{"s"},
+				Value:   descriptionNameSeparator,
+				Usage:   "separator that separate version name from version description in version description file",
+			},
+			&cli.PathFlag{
+				Name:     "inputDir",
+				Aliases:  []string{"id"},
+				Required: true,
+				Usage:    "path to versions folder",
+			},
+			&cli.PathFlag{
+				Name:    "outputFilepath",
+				Aliases: []string{"of"},
+				Usage:   "path to place generator output file",
+			},
+		},
+		Action: func(context *cli.Context) error {
+			var err error
+			outputFilepath := context.Path("outputFilepath")
+			if outputFilepath == "" {
+				outputFilepath, err = os.Executable()
+				if err != nil {
+					return err
+				}
+				outputFilepath = filepath.Join(filepath.Dir(outputFilepath), outputFilename)
+			}
+			vReader := verReader{
+				versionsDir:              context.Path("inputDir"),
+				descriptionFilename:      context.String("descFilename"),
+				descriptionNameSeparator: context.String("descNameSeparator"),
+			}
+			versions, err := vReader.readVersionsDir()
+			if err != nil {
+				return err
+			}
+			baseUrl := context.String("url")
+			if !strings.HasSuffix(baseUrl, "/") {
+				baseUrl += "/"
+			}
+			for i, ver := range versions {
+				versions[i].VersionFolderUrl = baseUrl + ver.Version + "/"
+			}
+			jsonVersions, err := json.Marshal(versions)
+			if err != nil {
+				return err
+			}
+			err = ioutil.WriteFile(outputFilepath, jsonVersions, 0644)
+			if err != nil {
+				return fmt.Errorf("save output file error: %v", err)
+			}
+			fmt.Printf("Description created Sucessfully! Versions counter: %d", len(versions))
+			return nil
+		},
+	}
+	err := app.Run(os.Args)
+	if err != nil {
+		panic(err)
+	}
+}
+
+type verReader struct {
+	versionsDir              string
+	descriptionFilename      string
+	descriptionNameSeparator string
+}
+
+func (vr verReader) readVersionsDir() ([]updaterini.ServData, error) {
+	versions, err := ioutil.ReadDir(vr.versionsDir)
+	versionsSDData := make([]updaterini.ServData, 0)
+	if err != nil {
+		return nil, fmt.Errorf("read versions dir error: %v", err)
+	}
+	for _, version := range versions {
+		if !version.IsDir() {
+			continue
+		}
+		_, err = updaterini.ParseVersion(version.Name())
```

Comment:
It's true, I used replace, how to handle this right, by go best practice?

Reply:
Just use `go get module_name@branch` (or `@commit_hash` instead of `@branch`) to update go.mod. Using `replace` should break `go install` from remote repository as far as I remember

Reply:
I use replace with my local path, to see changes instantly. Fix now, thanks!